src/main/java/Duke/Storage/TaskStorage.java

```diff
@@ -0,0 +1,166 @@
+    /**
+     * Parses the string argument as a Task.
+     * The string is expected to be in the format "[T|D|E],[0|1],<description>,<date>",
+     * where `[T|D|E]` represents the type of task and `[0|1]` represents the completion status.
+     *
+     * @param data The string to be parsed.
+     * @return The Task object.
+     * @throws FileFormatException If the string is not in the specified format.
+     */
+    private static Task parseTask(String data) throws FileFormatException {
+        String[] splitData = data.split(",", -1);
+        if (splitData.length != 4) {
+            throw new FileFormatException("Expected 4 values, got " + splitData.length);
+        }
+
+        int completionStatus;
+        try {
+            completionStatus = Integer.parseInt(splitData[1]);
+        } catch (NumberFormatException e) {
+            throw new FileFormatException(String.format("Unable to parse \"%s\" as an integer", splitData[1]));
+        }
+        if (!(completionStatus == 0 || completionStatus == 1)) {
+            throw new FileFormatException("Expected 0 or 1 for completion status, got " + completionStatus);
+        }
+
+        String type = splitData[0], description = splitData[2], date = splitData[3];
+        boolean isDone = completionStatus == 1;
+
+        try {
+            switch (type) {
+                case Todo.TASK_TYPE_ICON:
+                    if (date.length() > 0) {
+                        throw new FileFormatException("Unexpected date for task type todo");
+                    }
+                    return new Todo(description, isDone);
+                case Deadline.TASK_TYPE_ICON:
+                    return new Deadline(description, LocalDate.parse(date), isDone);
+                case Event.TASK_TYPE_ICON:
+                    return new Event(description, LocalDate.parse(date), isDone);
+            }
+        } catch (InvalidTaskException e) {
+            throw new FileFormatException("", e);
+        } catch (DateTimeParseException e) {
+            throw new FileFormatException(String.format("Unable to parse \"%s\" as a date", date));
+        }
+
+        throw new FileFormatException(String.format("Unrecognized task type \"%s\"", type));
+    }
+
+    /**
+     * Returns the specified task as a string.
+     * The string is in the format "[T|D|E],[0|1],<description>,<date>",
+     * where `[T|D|E]` represents the type of task and `[0|1]` represents the completion status.
+     *
+     * @param task The task to encode.
+     * @return The string representing the task.
+     */
+    private static String encodeTask(Task task) {
+        String type = task.getTaskTypeIcon(), completionStatus = task.isDone() ? "1" : "0",
+                description = task.getDescription(), date = "";
+
+        if (task instanceof Deadline) {
+            date = ((Deadline) task).getEndDate().toString();
+        }
+        if (task instanceof Event) {
+            date = ((Event) task).getAt().toString();
+        }
```

Comment:
Would it make sense to use `else if` here since a `Task` cannot be a `Deadline` and `Event` at the same time?